src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -143,6 +150,23 @@
         <TargetOutputPath>$(_BaseBlazorRuntimeOutputPath)%(FileName)%(Extension)</TargetOutputPath>
       </_BlazorOutputWithTargetPath>
 
+
+     <!--
+      Once we processed the remaining targets, we need to add in any lazy-loaded
+      components by looking through the set of _BlazorManagedRuntimeAssembly and
+      picking out the assemblies that are:
+      - Included in the list of _LazyLoadedAssemblies
+      - Have not already been added to the Blazor output because they were resolved or required elsewhere
+      - OR are a root assembly if we are running with the mono linker
```

Comment:
I'm confused about this logic!

> Have not already been added to the Blazor output because they were resolved or required elsewhere

Wouldn't *all* dependency assemblies typically be added to the Blazor output and hence by excluded by the 2nd condition? The only way they wouldn't be added to the output is if the application doesn't have any chain of references to them.

I may be missing something, but I would have thought that if we're auto-detecting which dependencies are lazy, then the approach would be something like "mark all the non-lazy assemblies, and whatever remains are the lazy ones", like the "mark" phase in GC. Otherwise I'm unsure how dependencies `App -> RCL1 (lazy) -> Dep` and `App -> RCL2 -> Dep` would be treated. Of course, I might be misunderstanding some aspect of the design!

More broadly, I wonder how useful it is to auto-detect the dependencies anyway. Is there a plan for the dependency path info to be available at runtime somehow? I had thought people would write runtime logic like this to trigger the loading of lazy assemblies:

```cs
async Task OnNavigating(object sender, NavigatingEventArgs args)
{
    if (args.Url.StartsWith("admin"))
    {
        // Here we have to list the full transitive set of dependencies required in this part of the app
        await AssemblyLoader.LoadAssembliesAsync(
            "MyCompany.MyApp.Admin",
            "ComponentsVendor.BlazorComponents",
            "Newtonsoft.Json",
            "System.Memory",
            "System.Data");
    }
    else if (args.Url == "Products/Export")
    {
        // The PDF exporter is big so we only load it on this one page
        await AssemblyLoader.LoadAssembliesAsync("ComponentsVendor.PdfExport");
    }
}
```

If this is the case, then the developer still has to hard-code knowledge of the full transitive set of dependencies in their runtime code, so I'm not certain how helpful it is to auto-detect aspects of it at build time.

We do have the option of doing something higher-level where the developer doesn't specify individual assemblies to be loaded at runtime, but instead specifies something like which top-level lazy-loaded thing is to be loaded, and the system uses knowledge of the dependency graph to get all the right things loaded. However that takes us pretty close back to the design we were trying to avoid because it's more sophisticated than the min-bar and is something that developers could build on their own if they needed.

I'd be interested to know what you think is preferable here. We *can* still do the more high-level solution if you think it's going to be pretty cheap to implement, or we could consider that a possible future enhancement with the initial version simply requiring the exact full set of assemblies to be specified both at build-time and runtime.